Subviews & MSVC:  "Use of undefined type GemRB::GUIFactory"

This error is really confusing me because I can't understand why it happens with Visual Studio and not GCC

The following files fail to build with the same error ```C2027, Use of undefined type GemRB::GUIFactory (Holder.h Line 76)```

```Animations,cpp, Calendar.cpp, AnimationFactory.cpp, Core.cppm ControlAnimation.cpp, DialogHandler.cpp, DisplayMessage.cpp```

```FAILED: gemrb/core/CMakeFiles/gemrb_core.dir/Animation.cpp.obj 
  B:\VS\VC\Tools\MSVC\14.26.28801\bin\HostX64\x64\cl.exe  /nologo /TP -DGEM_BUILD_DLL -DHAVE_CONFIG_H -Dgemrb_core_EXPORTS -I. -I..\..\..\gemrb\includes -I..\..\..\gemrb\core /DWIN32 /D_WINDOWS /W3 /GR /EHsc /MD /Zi /O2 /Ob1 /DNDEBUG /showIncludes /Fogemrb\core\CMakeFiles\gemrb_core.dir\Animation.cpp.obj /Fdgemrb\core\CMakeFiles\gemrb_core.dir\ /FS -c ..\..\..\gemrb\core\Animation.cpp
B:\msvc\src\chiv\gemrb\core\Holder.h(76): error C2027: use of undefined type 'GemRB::GUIFactory'
  B:\msvc\src\chiv\gemrb\core\Interface.h(78): note: see declaration of 'GemRB::GUIFactory'
  B:\msvc\src\chiv\gemrb\core\Holder.h(72): note: while compiling class template member function 'GemRB::Holder<GemRB::GUIFactory>::Holder(const GemRB::Holder<GemRB::GUIFactory> &)'
  B:\msvc\src\chiv\gemrb\core\Interface.h(819): note: see reference to function template instantiation 'GemRB::Holder<GemRB::GUIFactory>::Holder(const GemRB::Holder<GemRB::GUIFactory> &)' being compiled
  B:\msvc\src\chiv\gemrb\core\Interface.h(335): note: see reference to class template instantiation 'GemRB::Holder<GemRB::GUIFactory>' being compiled
```

To work around this, I looked for where Interface.h was including "Holder.h" and added an include for GUI/GUIFactory.h just before it, and then all those files build.


I just don't know if this is right or not, since there is too much stuff happening here that I don't understand. I did manage to run the game and walk around/use the gui though. (that is, after hacking around the final issue I had)

--- a/gemrb/core/Interface.cpp
+++ b/gemrb/core/Interface.cpp
@@ -72,7 +72,6 @@
 #include "GUI/Button.h"
 #include "GUI/Console.h"
 #include "GUI/EventMgr.h"
-#include "GUI/GUIFactory.h"
 #include "GUI/GameControl.h"
 #include "GUI/GUIScriptInterface.h"
 #include "GUI/Label.h"

--- a/gemrb/core/Interface.h
+++ b/gemrb/core/Interface.h
@@ -35,6 +35,7 @@
 #include "GameData.h"
 #include "GUI/Tooltip.h"
 #include "GUI/Window.h"
+#include "GUI/GUIFactory.h"
 #include "Holder.h"
 #include "ImageMgr.h"
 #include "InterfaceConfig.h"
@@ -74,7 +75,6 @@ class Font;
 class Game;
 class GameControl;
 class GlobalTimer;
-class GUIFactory;
 class ITMExtHeader;
 class Image;
 class Item;

